src/genai/schema/_api.py

```diff
@@ -142,603 +232,472 @@ class ModelType(str, Enum):
 
 
 class ModerationHAP(ApiBaseModel):
-    input: Optional[bool] = True
+    input: Optional[ModerationHAPInput] = None
+    output: Optional[ModerationHAPOutput] = None
+
+    def model_post_init(self, __context: Any) -> None:
+        # TODO: remove in next major release
+        deprecated_attrs = {
+            name: getattr(self, name) for name in ["threshold", "send_tokens"] if getattr(self, name, None) is not None
+        }
+        for attr in deprecated_attrs:
+            _log_deprecation_warning(
+                f"genai.schema.ModerationHAP.{attr}",
+                f"Property '{attr}' is deprecated! Use 'input' and 'output' property.",
+            )
+            self.input = self.input or ModerationHAPInput(enabled=True, **deprecated_attrs)
```

Comment:
does this actually work if I have
`ModerationHap(input=True, threshold=0.75)`

The field_validator for `input` will convert it to
`ModerationHAPInput(enabled=True)` so `input` on this line is truthy and the `deprecated_attrs` are not passed or am I missing something?

Reply:
It does not; I've oversimplified it. Could you check the implementation now?

Reply:
```python
    def model_post_init(self, __context: Any) -> None:
        deprecated_attrs = {
            name: getattr(self, name) for name in ["threshold", "send_tokens"] if getattr(self, name, None) is not None
        }
        for name, value in deprecated_attrs.items():
            _print_deprecation_warning(
                f"(ModerationStigma): '{name}' is deprecated! Use 'input' parameter (ModerationStigmaInput) 'output' parameter (ModerationStigmaOutput) instead.",
            )
            self.input = self.input or ModerationStigmaInput(enabled=True)
            setattr(self.input, name, value)

            self.output = self.output or ModerationStigmaOutput(enabled=True)
            setattr(self.output, name, value)

            delattr(self, name)

    @field_validator("input", mode="before")
    @classmethod
    def _validate_input(cls, value: Any):
        return ModerationStigmaInput(enabled=value) if isinstance(value, bool) else value
```

Won't this enable input if its `None`?
If I have 
`ModerationStigma(output=True, threshold=0.75)`,
looks like this line will enable input even if it was disabled:
`self.input = None or ModerationStigmaInput(enabled=True)`